Add npm-run-all package to simplify running scripts

mkaz opened the pull request:
## Description

The npm-run-all package allows for an easier way to run multiple scripts
from the package.json file. We are going to run a few scripts to
generate api docs, so instead of joining multiple commands using && it
can be simplified to creating a command for each and then running:

`npm-run-all api-docs:*`

Package docs:
https://www.npmjs.com/package/npm-run-all

## How has this been tested?

Run: `npm run docs:build` and confirm it runs the set of scripts specified.


## Types of changes

- Moves api doc generation scripts to own commands
- Move doc generation script to own command
- Uses npm-run-all in docs:build command to run others

mkaz committed:
Add npm-run-all package to simplify running scripts

The npm-run-all package allows for an easier way to run multiple scripts
from the package.json file. We are going to run a few scripts to
generate api docs, so instead of joining multiple commands using && it
can be simplified to creating a command for each and then running:

`npm-run-all api-docs:*`

Package docs:
https://www.npmjs.com/package/npm-run-all

gziolo commented:
We can also update this line with the newly added command:

https://github.com/WordPress/gutenberg/blob/8f90816cfe7c19baba2f5fc9a54c61b2eb59b2e3/package.json#L309

Now that we generate the docs for blocks based on `block.json` files, we should also add an entry in `lint-staged` that watches for changes in those files and triggers a re-run of the documentation for that specific aspect.

https://github.com/WordPress/gutenberg/blob/8f90816cfe7c19baba2f5fc9a54c61b2eb59b2e3/package.json#L290

mkaz committed:
Update lint-staged (pre-commit hook) with api-docs

Use the npm-run-all in the lint-staged command that gets run by husky on
pre-commit hooks. This allows for checking if any block.json and core
blocks are modified that triggers an update to the reference doc.

mkaz commented:
Good catch, the npm-run-all works there also, so just added the `api-docs:*` wilcard so any future additions will also get picked up by the pre-commit hook. 👍

gziolo commented on package.json:
```diff
@@ -306,7 +310,7 @@
 			"node ./docs/tool/index.js"
 		],
 		"packages/**/*.{js,ts,tsx}": [
```
We need to include JSON files as well 😄 
```suggestion
		"packages/**/*.{js,ts,tsx,json}": [
```

mkaz replied:
😁

mkaz committed:
Update package.json

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

gziolo reviewed (approved):
💯
